File: services/frontend/src/domains/project.ts
import type { DomainConfig, FieldConfig } from './base'
import type { GeneratedProjectAttrs } from './generated'
import { GENERATED } from './generated'

export type ProjectAttrs = GeneratedProjectAttrs

const FIELDS: FieldConfig<ProjectAttrs>[] = [
  { key: 'start_date', label: 'Start date', kind: 'date' },
  { key: 'target_end_date', label: 'Target end date', kind: 'date' },
  { key: 'completed_date', label: 'Completed date', kind: 'date' },
  { key: 'budget_estimate', label: 'Budget estimate', kind: 'number' },
]

export const projectConfig: DomainConfig<ProjectAttrs> = {
  label: 'Project',
  statuses: GENERATED.project.statuses,
  logTypes: GENERATED.project.logTypes,
  fields: FIELDS,
  defaultAttributes: () => ({ domain: 'project', related_entity_ids: [] }),
  namePlaceholder: 'Display name',
  locationPlaceholder: 'Garage, kitchen, ...',
  uiVariant: 'schedule',
}

File: libs/shared/src/aria_shared/models/entities/project.py
from datetime import date
from typing import ClassVar, Literal

from aria_shared.types import PyObjectId

from .base import BaseAttributes


class ProjectAttrs(BaseAttributes):
    DOMAIN: ClassVar[str] = "project"
    VALID_STATUSES: ClassVar[set[str]] = {"planning", "in_progress", "on_hold", "completed"}
    LOG_TYPES: ClassVar[set[str]] = {"service", "repair", "inspection", "expense", "note", "milestone"}

    domain: Literal["project"] = "project"
    related_entity_ids: list[PyObjectId] = []
    start_date: date | None = None
    target_end_date: date | None = None
    completed_date: date | None = None
    budget_estimate: float | None = None
